prowler/providers/azure/lib/service/service.py

```diff
@@ -5,13 +5,21 @@
 class AzureService:
     def __init__(
         self,
-        service: str,
+        services: list,
         audit_info: Azure_Audit_Info,
     ):
+        """
+        services must be a list of Azure clients to use in service (for now only GraphServiceClient support multiple client for use "v1" and "beta" versions)
+        If the service is not in a list, it will be converted to a list of one element
+        If a list is passed, and it not contains GraphServiceClient, it will be used only the last client of the list for all subscriptions
```

Comment:
```suggestion
        services argument must be a list of Azure client's class to be set for the current service (for now only GraphServiceClient supports multiple clients to use "v1" and "beta" versions)
        If the service not a list, thus a class object, it will be converted to a list of one element with that class object.
        If a list is passed, and it not contains GraphServiceClient, it will be used only the last client of the list for all subscriptions
```

Reply:
For the last sentence `If a list is passed, and it not contains GraphServiceClient, it will be used only the last client of the list for all subscriptions`. I think we should set up all the clients passed in the list instead of just setting the first or the last, what do you think?

Reply:
The first time that I was developing this I setup to store in a subscription multiple clients service if it is passed, the problem is it breaks all Azure services done by now. This is because all services assume that there is only one client per subscription.

For example this is app_insights service when it search for components the for loop is in subscriptions but it doesn't iterate over clients again because it assume that can only be one.
```
def __get_components__(self):
        logger.info("AppInsights - Getting components...")
        components = {}

        for subscription_name, client in self.clients.items():
            try:
                components_list = client.components.list()
                components.update({subscription_name: {}})

                for component in components_list:
                    components[subscription_name].update(
                        {
                            component.app_id: Component(
                                resource_id=component.id, resource_name=component.name
                            )
                        }
                    )
            except Exception as error:
                logger.error(
                    f"Subscription name: {subscription_name} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                )

        return components
```